### PR TITLE
Mark resetProperties and resetQpProperties as final 

### PR DESCRIPTION
**Pull request by sapitts**
### Design information

Follow on to PR #8030 to mark the `resetProperties` and `resetQpProperties` functions as `final` to prevent inheriting classes from overwriting these functions now that MAMMOTH has been updated.

@aeslaughter  or @bwspenc  please review

Refs #5481

### CODE REVIEW

**Review by bwspenc — APPROVED** [review]
Looks good to me.  Let's get Swetha's change merged in to BISON to fix those tests before we merge this in.

**Comment by sapitts** [comment]
@bwspenc  or @dschwen This is past testing and ready to merge when you have a chance.  thanks

**Pull request merged**
